### DISCUSSION TITLE
Input and buttons should not use max min or hight properties

### DISCUSSION THREAD

**Discussion by ashfahan** [saadeghi/daisyui#830] [Q&A] [answered]
Input and buttons should not use max min or hight properties because it make them hard to customize and instead we can use padding that would make them easily customizable and wider implimentation

**Comment by saadeghi** [comment, marked as answer]
In a design system, buttons have specific height values and that's the whole point of having different button sizes.  

If buttons don't have height and just stretch based on the content, it will cause design problems.

Imagine having 4 medium size buttons side by side:  
- one includes a single word text
- second one has a large SVG icon
- third one has a `<br>` inside
- fourth one has text with a tall font

These 4 buttons should have equal heights but instead each one will end up with a different height. That's why having a height value is required.

**Reply by ashfahan** [reply]
if thats the case there still should have been some way to configure it globally like a css variable be the one ?

**Reply by saadeghi** [reply]
There are 3 ways:  
1. You can use `h-auto` utility class to make a specific button `height:auto`:
  https://play.tailwindcss.com/QGn7z7xjuE  
    ```html
    <button class="btn h-auto">
      <img src="https://api.lorem.space/image/furniture?w=150&h=150">
    </button>
    ```
2. You can add a global CSS for all buttons:  
  https://play.tailwindcss.com/DRnNlsSdlD?file=css
    ```css
    .btn{
      height: auto;
    }
    ```
3. You can add a global CSS for a specific theme:
  https://play.tailwindcss.com/3rIpAu977S?file=config  
  Here I extend the built-in `light` theme and add a custom CSS to it:
    ```js
    daisyui: {
      themes: [
        {
          light: {
            ...require("daisyui/src/colors/themes")["[data-theme=light]"],
            ".btn": {
              "height": "auto",
            },
          },
        },
      ],
    },
    ```

**Reply by ashfahan** [reply]
i understand your point but actually the problem i had originated from inputs fields and i start questioning every where (form elements) why use height 

i wanted to modify i.e how small is small and how large is large e.g 0.75em or 0.75% of orginal for small and  1.25em or 1.25% for large ro something like that but i can see daisy have different design system than em or % .. either way there should have been some css variables or something that should define the small and large quantity and also something that can define the "normal" value

Coming back to buttons, I believe the UI is broken in their case. For example, if you look at this [example](https://play.tailwindcss.com/xMXXec9rGw), you can see even after h-auto button spacing on top and bottom is broken, but in the case of 'p-4 h-auto,' it appears to be the expected result.

**Reply by saadeghi** [reply]
Customizing sizes is not impossible. I gave you examples of how to do it. However, it's not using CSS variables.  
I agree that it would be cool to do it using CSS variables. Actually this is something I'm planning to do for the next major version but it takes a lot of refactoring since it's not just about buttons. Other elements have the same size values too.
